File: src/main/java/com/ftn/service/CompanyService.java
package com.ftn.service;


import com.ftn.model.Company;

import java.util.List;

/**
 * Created by Jasmina on 21/05/2017.
 */
public interface CompanyService {

    List<Company> read();

    Company read(Long id);

    Company create(Company company);

    Company update(Long id, Company company);

    void delete(Long id);
}

File: src/main/webapp/service/companyService.js
/**
 * Created by Alex on 5/30/17.
 */
app.service('companyService', function($http){
    return {
        read: function(onSuccess, onError){
            $http.get('/api/companies').then(onSuccess, onError);
        },
        readById: function(id, onSuccess, onError){
            $http.get('/api/companies/' + id).then(onSuccess, onError);
        },
        create: function(company, onSuccess, onError){
            $http.post('/api/companies', company).then(onSuccess, onError);
        },
        update: function(company, onSuccess, onError){
            $http.patch('/api/companies/' + company.id, company).then(onSuccess, onError);
        },
        delete: function (id, onSuccess, onError) {
            $http.delete('/api/companies/' + id).then(onSuccess, onError);
        }
    }
});